[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85161
* #84276
* __->__ #84275
* #84274

As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

Commit:
[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`

As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

slgong-fb commented on torch/csrc/profiler/collection.cpp:
```diff
@@ -805,7 +803,6 @@ trace_ptr_t addKinetoEvents(
   }
 
   auto trace = std::make_unique<ActivityTraceWrapper>(stopTrace());
-  TORCH_INTERNAL_ASSERT(trace || !kKinetoAvailable);
```
no replacement for this line? Just removing a assert?

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Cleanup collection.cpp and replace some asserts with `SOFT_ASSERT`"


As part of the reliability effort, I am pulling `TORCH_INTERNAL_ASSERT`s out of post processing and replacing them with `SOFT_ASSERT`s so we can gracefully recover and not crash training jobs when profiling is triggered.

Differential Revision: [D39108643](https://our.internmc.facebook.com/intern/diff/D39108643/)

[ghstack-poisoned]

robieta commented:
Putting these two on hold since https://github.com/pytorch/pytorch/pull/83965 is already reverted internally so raising on soft assert will be problematic once that revert syncs out.
